bootstrap/src/main/java/io/airlift/bootstrap/LifeCycleManager.java

```diff
@@ -40,6 +40,16 @@
     private final LifeCycleMethodsMap methodsMap;
     private final AtomicReference<Thread> shutdownHook = new AtomicReference<>();
 
+    /**
+     * Provides a mechanism to handle exceptions raised from {@link PreDestroy} methods in
+     * whatever way makes the most sense. Implementations should not raise exceptions of
+     * their own
+     */
+    public interface LifeCycleStopFailureHandler
```

Comment:
Let's make this private for now. We can make it public later if there is a use case.